include/qpid/dispatch/tls.h

```diff
@@ -0,0 +1,183 @@
+#ifndef __tls_h__
+#define __tls_h__ 1
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**@file
+ * Management of TLS configuration and state
+ */
+
+
+#include "qpid/dispatch/log.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+
+typedef struct qd_tls_config_t   qd_tls_config_t;    // run-time TLS configuration state
+typedef struct qd_tls_session_t  qd_tls_session_t;  // per connection TLS state
+typedef struct qd_ssl2_profile_t qd_ssl2_profile_t;  // sslProfile configuration record
+
+// Proton has two different TLS implementations: one for AMQP and a buffer-based one for use with Raw Connections:
+typedef enum {
+    QD_TLS_TYPE_NONE = 0,     // unset
+    QD_TLS_TYPE_PROTON_AMQP,  // for use with AMQP transport
+    QD_TLS_TYPE_PROTON_RAW,   // use raw connection/qd_buffer_t interface
+} qd_tls_type_t;
+
+typedef enum {
+    QD_TLS_CONFIG_MODE_NONE = 0,  // unset
+    QD_TLS_CONFIG_SERVER_MODE,    // Operate as a TLS server (i.e. listener socket)
+    QD_TLS_CONFIG_CLIENT_MODE,    // Operate as an TLS client (i.e. outgoing connections)
+} qd_tls_config_mode_t;
+
+// sslProfile configuration record
+struct qd_ssl2_profile_t {
```

Comment:
I am guessing `qd_ssl2_profile_t` will be renamed to `qd_ssl_profile_t` once the old `qd_ssl_profile_t` is removed ?

Reply:
Yes - I can do that on a follow on patch.